docs/environment-variables.adoc

```diff
@@ -51,6 +52,38 @@ Method-level configurations are visible in both `@BeforeEach` setup methods and
 Since v1.7.0, a class-level configuration means that the specified environment variables are cleared/set before and reset after each individual test in the annotated class.
 ====
 
+== `@RestoreEnvironmentVariables`
+`@RestoreEnvironmentVariables` can be used to restore changes to environment variables made directly in code.
+While `@ClearEnvironmentVariable` and `@SetEnvironmentVariable` set or clear specific variables and values, they don't allow values to be calculated or parameterized, thus there are times you may want to directly set them in your test code.
+`@RestoreEnvironmentVariables` can be placed on test methods or test classes and will completely restore all environment variables to their original state after a test or test class is complete.
+
+In this example, `@RestoreEnvironmentVariables` is used on a test method, ensuring any changes made in that method are restored:
+
+[source,java,indent=0]
+----
+include::{demo}[tag=environment_method_restore_test]
+----
+
+[NOTE]
+====
+Modifying environment variables of a running JVM requires several lines of code and the Java reflection API.
+The two `@RestoreEnvironmentVariables` examples leave out that detail and use a hypothetical `setEnvVar()` method.
+====
+
+When `@RestoreEnvironmentVariables` is used on a test class, any environment variables changes made during the entire lifecycle of test class, including test methods, `@BeforeAll`, `@BeforeEach` and 'after' methods, is restored after the test class lifecycle is complete.
+In addition, the annotation is inherited by each test method just as if each one was annotated with `@RestoreEnvironmentVariables`.
+
+In the following example, `SomeOtherTest` is completely isolated from changes made in `MyEnvironmentVariableRestoreTest` and `isolatedTest2` is isolated from changes made in `isolatedTest1`.
+Both `isolatedTest1` and `isolatedTest2` _will_ see environment variable values set in `@BeforeAll` and `@BeforeEach`.
+
+[source,java,indent=0]
+----
+include::{demo}[tag=environment_class_restore]
+----
+
+Note that using `@RestoreEnvironmentVariables` is not necessary to restore environment variables modified via `@ClearEnvironmentVariable` or `@SetEnvironmentVariable`.
+However, it is OK and normal to use all three annotations together, which you might do if some environment values are parameterized (i.e. need to be set in code) and others are not.
```

Comment:
Late addition to the docs:  The potential use of the three annotations together.

Reply:
I suggest to add an example to this too. It might be clear for most users, but I think it might also help not that experienced ones :)

Reply:
I did add an example for the three annotation together.